src/components/structures/RoomView.js

```diff
@@ -747,6 +750,18 @@ export default createReactClass({
                 });
             }
         }
+        if (!SettingsStore.getValue('dontShowChatEffects')) {
+            this.context.on('Event.decrypted', this.onEventDecrypted);
```

Comment:
this looks like it'll leak event handlers very swiftly

Reply:
☹️ could you please tell me why is that?. i just pushed my latest changes to branch. could you please check why still on the first refresh i get the confetti animation again. "i guess after the first time it is not fun anymore :P 😅"

Reply:
The handler should be created/destroyed once at a higher level like the constructor so that you don't add the handler multiple times